libcodechecker/server/server.py

```diff
@@ -329,13 +330,11 @@ def do_POST(self):
                     elif request_endpoint == 'CodeCheckerService':
                         # This endpoint is a product's report_server.
                         if not product:
-                            LOG.debug("Requested CodeCheckerService on a "
-                                      "nonexistent product.")
-                            self.send_error(  # 404 Not Found
-                                404,
-                                "The specified product '{0}' does not exist!"
-                                .format(product_endpoint))
-                            return
+                            error_msg = "Requested CodeCheckerService on a " \
+                                        "nonexistent product." \
+                                        .format(product_endpoint)
```

Comment:
This string literal doesn't contain a `{}` to replace with the product endpoint.